src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildLazyLoadTest.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+using static Microsoft.AspNetCore.Components.WebAssembly.Build.WebAssemblyRuntimePackage;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+{
+    public class BuildLazyLoadTest
+    {
+        [Fact]
+        public async Task Build_LazyLoadExplicitAssembly_Debug_Works()
+        {
+            // Arrange
+            using var project = ProjectDirectory.Create("standalone", additionalProjects: new[] { "razorclasslibrary" });
+            project.Configuration = "Debug";
+
+            project.AddProjectFileContent(
+@"
+<ItemGroup>
+    <BlazorLazyLoad Include='RazorClassLibrary.dll' />
+</ItemGroup>
+");
+
+            var result = await MSBuildProcessManager.DotnetMSBuild(project);
+
+            var buildOutputDirectory = project.BuildOutputDirectory;
+
+            // Verify that a blazor.boot.json file has been created
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "blazor.boot.json");
+            // And that the assembly is in the output
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "RazorClassLibrary.dll");
+
+            var bootJson = ReadBootJsonData(result, Path.Combine(buildOutputDirectory, "wwwroot", "_framework", "blazor.boot.json"));
+
+            // And that it has been labelled as a dynamic assembly in the boot.json
+            var dynamicAssemblies = bootJson.resources.dynamicAssembly;
+            var assemblies = bootJson.resources.assembly;
+
+            Assert.NotNull(dynamicAssemblies);
+            Assert.Contains("RazorClassLibrary.dll", dynamicAssemblies.Keys);
+            Assert.DoesNotContain("RazorClassLibrary.dll", assemblies.Keys);
+
+            // App assembly should not be lazy loaded
+            Assert.DoesNotContain("standalone.dll", dynamicAssemblies.Keys);
+            Assert.Contains("standalone.dll", assemblies.Keys);
+        }
+
+        [Fact]
+        public async Task Build_LazyLoadExplicitAssembly_Release_Works()
```

Comment:
Could you also add a test for release? I know it's immaterial here, but it's good to have for the SDK

Reply:
Sure -- just added them.